Assets/MixedRealityToolkit.Examples/Experimental/SceneUnderstanding/Scripts/DemoSceneUnderstandingController.cs

```diff
@@ -0,0 +1,395 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.Experimental.SpatialAwareness;
+using Microsoft.MixedReality.Toolkit.UI;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using TMPro;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Examples
+{
+    public class DemoSceneUnderstandingController : MonoBehaviour, IMixedRealitySpatialAwarenessObservationHandler<SpatialAwarenessSceneObject>
+    {
+        public string SavedSceneNamePrefix = "DemoSceneUnderstanding";
+        public GameObject StuffToPlace;
+        public bool InstantiatePrefabs;
+        public GameObject InstantiatedPrefab;
+        public Transform InstantiatedParent;
+        [Header("UI")]
+        public GameObject autoUpdateToggle;
+        public GameObject quadsToggle;
+        public GameObject meshesToggle;
+        public GameObject maskToggle;
+        public GameObject platformToggle;
+        public GameObject wallToggle;
+        public GameObject floorToggle;
+        public GameObject ceilingToggle;
+        public GameObject worldToggle;
+        public GameObject usePersistentToggle;
+        public GameObject completelyToggle;
+        public GameObject backgroundToggle;
+
+        List<SpatialAwarenessSceneObject> platforms = new List<SpatialAwarenessSceneObject>(16);
+
+        IMixedRealitySpatialAwarenessSceneUnderstandingObserver observer;
+
+        bool isInit = false;
+
+        // by default stick something on the nearest platform without the user requesting it
+        bool autoPlaceStuffOnce = true;
```

Comment:
nit: mark all of these as private or protected to keep with convention of codebase